alea/test/stream_serializer.cpp

```diff
@@ -0,0 +1,77 @@
+#include <alps/alea/plugin/stream_serializer.hpp>
+
+#include <alps/testing/near.hpp>
+#include "gtest/gtest.h"
+#include "dataset.hpp"
+
+#include <vector>
+#include <sstream>
+
+// Emulate Boost.Serialization insertion/extraction operators
+template<typename R>
+typename std::enable_if<alps::alea::is_alea_result<R>::value, std::stringstream &>::type
+operator<<(std::stringstream &ss, const R &result) {
+    save(ss, result, 0);
+    return ss;
+}
+
+template<typename R>
+typename std::enable_if<alps::alea::is_alea_result<R>::value, std::stringstream &>::type
+operator>>(std::stringstream &ss, R &result) {
+    load(ss, result, 0);
+    return ss;
+}
+
+template<typename T>
+std::stringstream & operator<<(std::stringstream &ss, const std::vector<T> &v) {
+    for(auto const& x : v) ss << x;
+    return ss;
+}
+
+template<typename T>
+std::stringstream & operator>>(std::stringstream &ss, std::vector<T> &v) {
+    for(auto & x : v) ss >> x;
+    return ss;
+}
+
+template <typename Acc>
+class twogauss_serialize_case
+    : public ::testing::Test
+{
+public:
+    typedef typename alps::alea::traits<Acc>::value_type value_type;
+    typedef typename alps::alea::traits<Acc>::result_type result_type;
+
+    twogauss_serialize_case() { }
+
+    void test_result()
+    {
+        Acc in_acc(2);
+        for (size_t i = 0; i != twogauss_count; ++i)
+            in_acc << std::vector<double>{twogauss_data[i][0], twogauss_data[i][1]};
+        auto in = in_acc.result();
+
+        std::stringstream ss;
+        ss << in; // serialize
+
+        Acc out_acc(2);
+        auto out = out_acc.result();
+
+        ss >> out; // deserialize
+
+        // FIXME: How do I check that in == out in the generic way?
+        std::cout << ss.str() << std::endl;
```

Comment:
We should provide an util for this.  One needs to be a little bit careful here though with a strict `==` comparison, because the stream you write to might be a text file and then you may very well lose a bit-by-bit equivalence after reading it back.  Probably something like Eigen's `isApprox()`, I'm thinking something along the lines:

    template <typename Res>
    alps::alea::util::is_approx(const Res &a, const Res &b, double atol);

Reply:
If we are to test serialization/deserialization, the comparison must be strict. Of course, different archive implementations may allow loss of precision. But in this particular test we need an archive implementation/emulation that is guaranteed to store the unaltered data (`std::stringstream` is probably not good enough for this...)

With this said, having `is_approx` would be nice indeed.